itou/utils/sync.py

```diff
@@ -1,14 +1,33 @@
 import json
+from dataclasses import dataclass
+from enum import Enum, auto
+from typing import Callable, Union
 
 from django.db.models import QuerySet
 
 
+class DiffItemKind(Enum):
+    ADDITION = auto()
+    DELETION = auto()
+    EDITION = auto()
+    SUMMARY = auto()
+
+
+@dataclass
+class DiffItem:
+    key: str
+    kind: DiffItemKind
+    label: str
+    raw: dict = None
+    db_pk: Union[str, int] = None  # database pk in the case of an update
+
+
 def yield_sync_diff(
     collection: list[dict],
     collection_key: str,
     queryset: QuerySet,
     queryset_key: str,
-    compared_keys: list[tuple[str, str]],
+    compared_keys: list[tuple[Union[str, Callable], str]],
```

Comment:
Je me demande si ça fonctionne ça
```suggestion
    compared_keys: list[tuple[str | Callable, str]],
```